fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/SubqueryOperator.java

```diff
@@ -7,33 +7,37 @@
 import com.starrocks.sql.ast.QueryStatement;
 import com.starrocks.sql.optimizer.base.ColumnRefSet;
 import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.logical.LogicalApplyOperator;
+import com.starrocks.sql.optimizer.transformer.OptExprBuilder;
 
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 
 public class SubqueryOperator extends ScalarOperator {
 
-    // mark work way
-    private boolean useSemiAnti;
     private final QueryStatement queryStatement;
+    private final LogicalApplyOperator applyOperator;
+    private final OptExprBuilder rootBuilder;
```

Comment:
why LogicalApplyOperator and OptExprBuilder could be SubqueryOperator class field?

Reply:
The process of subquery transform can be described in two steps

1. Parse subquery to ApplyOperator while transforming Expr to ScalarOperator, but without attaching to logical plan, only save the applyOperator and rootBuilder to the `SubqueryOperator` for later usage.
2. Attach the remaining ApplyOperator(from the `SubqueryOperator`'s field) to the logical plan after all the scalar rewritten rules are performed.

Reply:
OptExprBuilder can be replaced with OptExpression if the second step is executed in the optimization stage(next pr)